library/public/interactor.h

```diff
@@ -65,77 +116,89 @@ class F3D_EXPORT interactor
+   * Use `getBindingDocumentation` to access this doc.
+   *
+   * When the corresponding bind happens, the provided commands will be triggered using
+   * triggerCommand. Considering checking if an interaction exists or removing it before adding it
+   * to avoid potential conflicts.
    *
    * ANY modifier interactions will only be triggered if no other interaction bind with modifier
    * is found.
    *
-   * Adding commands for an existing combination of interaction and modifier will throw a
-   * interactor::already_exists_exception.
+   * Adding commands for an existing bind will throw a interactor::already_exists_exception.
    */
-  virtual interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::vector<std::string> commands) = 0;
+  virtual interactor& addBinding(const interaction_bind_t& bind, std::vector<std::string> commands,
+    std::string group = std::string(),
+    std::function<std::pair<std::string, std::string>()> documentationCallback = nullptr) = 0;
 
   /**
    * See addBinding
-   * Convenience method to add a single command for an interaction, similar as
-   * addBinding(interaction, modifiers, {command})
+   * Convenience method to add a single command for an interaction,
+   * similar as `addBinding(bind, {command})`
    *
-   * Adding command for an existing combination of interaction and modifier will throw a
-   * interactor::already_exists_exception.
+   * Adding command for an existing bind will throw a interactor::already_exists_exception.
    */
-  virtual interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::string command) = 0;
+  virtual interactor& addBinding(const interaction_bind_t& bind, std::string command,
+    std::string group = std::string(),
+    std::function<std::pair<std::string, std::string>()> documentationCallback = nullptr) = 0;
 
   /**
    * Convenience initializer list signature for add binding method
    */
-  interactor& addBinding(
-    const std::string& interaction, ModifierKeys modifiers, std::initializer_list<std::string> list)
+  interactor& addBinding(const interaction_bind_t& bind, std::initializer_list<std::string> list,
+    std::string group = std::string(),
+    std::function<std::pair<std::string, std::string>()> documentationCallback = nullptr)
   {
-    return this->addBinding(interaction, modifiers, std::vector<std::string>(list));
+    return this->addBinding(bind, std::vector<std::string>(list), group, documentationCallback);
   }
 
   /**
    * Remove binding corresponding to provided interaction and modifiers
    */
-  virtual interactor& removeBinding(std::string interaction, ModifierKeys modifiers) = 0;
+  virtual interactor& removeBinding(const interaction_bind_t& bind) = 0;
+
+  /**
+   * Return a vector of available bind groups, in order of addition
+   */
+  virtual std::vector<std::string> getBindGroups() const = 0;
```

Comment:
Since we have `addBinding`/`removeBinding`, I suppose it makes more sens to call this function
```suggestion
  virtual std::vector<std::string> getBindingGroups() const = 0;
```